src/trainprogram.cpp

```diff
@@ -317,7 +320,9 @@ double trainprogram::TimeRateFromGPX(double gpxsecs, double videosecs, double cu
         double avgSpeedForLimit = avgSpeedFromGpxStep(currentStep + 1, 5);
         if (avgSpeedForLimit > 0.0) {
             bike *dev = (bike *)bluetoothManager->device();
-            dev->setSpeedLimit(avgSpeedForLimit * 3);
+            // bepo70: Video is recorded with Factor 2, Replay allows Factor 2 max, so set the speed Limit to 4 times speed to
```

Comment:
what if the video was recorded with a classic factor of 1? I mean we should probably implement something generic. What do you think?

Reply:
Totally agreed. I have added the Recording Factor in the calculation of maximum Speed allowed.

Reply:
thanks, but now you will have a different return on your code since, the default recording factor is 1 not 2. Maybe we should get the recording factor from a tag in the gpx file? what do you think?

Reply:
no, i have the same return because my videos have recording factor 2. The Recording factor is already calculated by the ration of gpx -length to video length, so there is no need to extra save it in the gpx file.

Reply:
sorry, you're right!